### PR TITLE
Abnormal hr output

### PR DESCRIPTION
**Pull request by sebjshaw**
Purpose of this pull request:
To give Dom an up to date version of the kafka consumer and SQLite database to allow him to populate the dashboard

**Added a file called abnormal-hr**

- This calculates the maximum heart rate of the user based on ages (still need to include weight into this calculation)
- Compares every heart rate against this maximum 

**Still needed**

- The code from the Lambda which talks to the SNS topic is needed to 
- Create the JSON that will be used to construct the personalised email

### CODE REVIEW

File: ec2-dash/dash_sqlite.py [hunk 1]
```diff
@@ -185,18 +190,30 @@ def push_to_s3():
...
+                # set new max_hr
+                # max_hr = calculate_max_heart_rate(log_entry['date_of_birth'])
                 
 
             # only adds to the database if it is a full entry      
             if len(log_entry) == 7:
-                add_entry_to_table(cursor, conn, log_entry)
+                add_entry_to_table(cursor, conn, log_entry, ride_id)
+                # if compare_hr_to_max_hr(log_entry['heart_rate'], max_hr) == True:
+                    # send_user_info_to_lambda(log_entry['heart_rate'], log_entry['duration'])
```
Comment (DomLaw82) [inline ec2-dash/dash_sqlite.py:217]: Like the max hr calculation setup, could we pas this to the dashboard as part of the user info dict to include a visual representation of the max on the hr graph?

File: ec2-dash/dash_sqlite.py [hunk 2]
```diff
@@ -185,18 +190,30 @@ def push_to_s3():
...
+
+                #regenerate the tables
+                recreate_current_ride_table(cursor, conn)
+                recreate_user_info_table(cursor, conn, log_entry)
+
+                # regenerate the join_ids and user_info csvs
+                ride_id = recreate_ride_id_from_datetime(log_entry)
+                ride_date = log_entry['date'] + " " + log_entry['time']
+                recreate_join_csv(log_entry['user_id'], ride_id)
+
```
Comment (DomLaw82) [inline ec2-dash/dash_sqlite.py:207]: For the ride csv, we are including the user id and ride id along with the aggregated data for each ride as one row, so the separate join table is no longer needed

**Review by DomLaw82 — APPROVED** [review]

File: ec2-dash/dash_sqlite.py (hunk 1 above)
Comment (sebjshaw) [reply ec2-dash/dash_sqlite.py:249]: I like this idea, shifted the code round and the user_info now includes max_hr as the last entry

File: ec2-dash/dash_sqlite.py (hunk 2 above)
Comment (sebjshaw) [reply ec2-dash/dash_sqlite.py]: Also done. Only two csv files are now created and pushed to s3, the user info and the log info for the most recent ride

**Pull request merged**